pennylane/qaoa/cost.py

```diff
@@ -244,3 +245,217 @@ def maxcut(graph):
+        applied to the :math:`i`-th vertex.
+
+        The returned mixer Hamiltonian is :func:`~qaoa.bit_flip_mixer` applied to :math:`G`.
+
+        .. note::
+
+            **Recommended initialization circuit:**
+                Each wire in the :math:`|1\rangle` state
+
+        **Unconstrained**
+
+        The Minimum Vertex Cover cost Hamiltonian for unconstrained QAOA is defined as:
+
+        .. math:: H_C \ = \ \sum_{(i, j) \in E(G)} (Z_i Z_j \ + \ Z_i \ + \ Z_j) \ - \
+                  \displaystyle\sum_{i \in V(G)} Z_i
+
+        where :math:`E(G)` is the set of edges of :math:`G`, :math:`V(G)` is the set of vertices,
+        and :math:`Z_i` is the Pauli-Z operator acting on the :math:`i`-th vertex.
+
+        The returned mixer Hamiltonian is :func:`~qaoa.x_mixer` applied to all wires.
+
+        .. note::
+
+            **Recommended initialization circuit:**
+                Even superposition over all basis states
+
+    """
+
+    if not isinstance(graph, nx.Graph):
+        raise ValueError("Input graph must be a nx.Graph, got {}".format(type(graph).__name__))
+
+    if constrained:
+        return (bit_driver(graph.nodes, 0), qaoa.bit_flip_mixer(graph, 1))
+
+    cost_h = edge_driver(graph, ["11", "10", "01"]) + bit_driver(graph.nodes, 0)
+    mixer_h = qaoa.x_mixer(graph.nodes)
+
+    return (cost_h, mixer_h)
+
+
+def max_clique(graph, constrained=True):
+    r"""Returns the QAOA cost Hamiltonian and the recommended mixer corresponding to the Maximum Clique problem,
+    for a given graph.
+
+    The goal of Maximum Clique is to find the largest `clique <https://en.wikipedia.org/wiki/Clique_(graph_theory)>`__ of a
+    graph --- the largest subgraph such that all vertices are connected by an edge.
+
+    Args:
+        graph (nx.Graph): a graph whose edges define the pairs of vertices on which each term of the Hamiltonian acts
+        constrained (bool): specifies the variant of QAOA that is performed (constrained or unconstrained)
+
+    Returns:
+        (.Hamiltonian, .Hamiltonian): The cost and mixer Hamiltonians
+
+    .. UsageDetails::
+
+        There are two variations of QAOA for this problem, constrained and unconstrained:
+
+        **Constrained**
+
+        .. note::
+
+            This method of constrained QAOA was introduced by Hadfield, Wang, Gorman, Rieffel, Venturelli, and Biswas
+            in `[arXiv:1709.03489] <https://arxiv.org/abs/1709.03489>`__.
+
+        The Maximum Clique cost Hamiltonian for constrained QAOA is defined as:
+
+        .. math:: H_C \ = \ \displaystyle\sum_{v \in V(G)} Z_{v}
+
+        where :math:`V(G)` is the set of vertices of the input graph, and :math:`Z_i` is the Pauli-Z operator
+        applied to the :math:`i`-th
+        vertex.
+
+        The returned mixer Hamiltonian is :func:`~qaoa.bit_flip_mixer` applied to :math:`\bar{G}`
+        (the complement of the graph).
```

Comment:
```suggestion
        The returned mixer Hamiltonian is :func:`~qaoa.bit_flip_mixer` applied to :math:`\bar{G}`,
        the complement of the graph.
```
Unnecessary parenthesis, +1 cent for me :moneybag: